### PR TITLE
Update to ESP-IDF 5.1 and add C6 and H2 support

### PR DESCRIPTION
**Pull request by tannewt**
* H2 is the first ESP without wifi.
* Code size grew so 2MB and 4MB C3s have `_bleio` disabled to make plenty of room.
* GCC was updated so some changes are to fix new compiler errors.
* adc and dac drivers were updated to new API. (C6 only supports new adc API.)
* C6 and H2 have a new RTC peripheral, LP_AON, that is used in place of the RTC on those chips.

### CODE REVIEW

File: ports/espressif/mpconfigport.mk [hunk 1]
```diff
@@ -2,7 +2,7 @@
 INTERNAL_FLASH_FILESYSTEM = 1
 
 # Internal math library is substantially smaller than toolchain one
-INTERNAL_LIBM = 1
+INTERNAL_LIBM = 0
```
Comment (dhalbert) [inline ports/espressif/mpconfigport.mk:5]: Why did you change this to use the ESP-IDF libm?

**Review by dhalbert — APPROVED** [review]
Thanks for bringing us up to date!

Approving, but one q about `INTERNAL_LIBM`. Merge when you wish.

File: ports/espressif/mpconfigport.mk (hunk 1 above)
Comment (tannewt) [reply ports/espressif/mpconfigport.mk:5]: There is a rom blob (can't remember which) that links against `floor()`. Our implementation has it for doubles but it didn't compile. So, I switched it assuming that's what espressif uses.
